### DISCUSSION TITLE
I can use rq and redis without first starting regis-server

### DISCUSSION THREAD

**Discussion by fcoder** [rq/rq#1856] [Q&A]
Hi,
Your document says that in order to use Python packages rq and regis, we have to run redis-server first, but I noticed that on my Macbook in a virtual environment for an Flask app, I don't need to start redis-server manually, it seems that the following rq command will start redis-server for me,
     rq workers <queue-name>

This can be verified by the following steps:

(1) Reboot the MacBook, don't start "redis-server"

(2) Execute command "netstat | grep 6379" to confirm that redis-server is not running.  See nothing as expected.

(3) Execute command "rq workers <queue-name>"

(4) Execute command "netstat | grep 6379" to check if redis-server is running, I see,
      ~% netstat | grep 6379
      tcp6       0      0  localhost.6379         localhost.49213        ESTABLISHED
      tcp6       0      0  localhost.49213        localhost.6379         ESTABLISHED
      tcp6       0      0  localhost.6379         localhost.49212        ESTABLISHED
      tcp6       0      0  localhost.49212        localhost.6379         ESTABLISHED
      so it seems "rq workers <queue-name> has invoked redis-server

(5) I can use the Redis with worker and queue without ever manually starting redis-server.

Can anybody confirm that starting redis-server before using Python rq and redis packages is not required anymore?

The versions that I used are as follows
redis               4.1.2
rq                  1.10.1
~% redis-server -v
Redis server v=7.0.7

Thanks

**Comment by ccrvlh** [comment]
Can you do `brew services list | grep redis` on Step 2 and check that your Redis server is actually `stopped`?

I'm saying this becauase I'm not sure if `netstat` is showing everything that is listening on 6379 or only connected - my Redis is currently `started` (as per brew command), but my `netstat` is also empty - so it's not a reliable way to see if the server is actually running.

`lsof` howerver shows that the Redis server is listening on 6379: `lsof -i -P | grep LISTEN | grep 6379`

The worker shouldn't be starting a server manually.

**Reply by fcoder** [reply]
@lowercase000, 
Thank you for your quick reply.  I just rebooted the MacBook then ran the two commands you suggested.  Yes, redis was indeed running right after MacBook rebooted and port 6379 was being listened to, netstat still showed nothing,  See screen dump below,

~% brew services list | grep redis
redis         started         Yuming ~/Library/LaunchAgents/homebrew.mxcl.redis.plist

~% lsof -i -P | grep LISTEN | grep 6379
redis-ser 649 Yuming    6u  IPv4 0x10d06743948e5a9b      0t0  TCP localhost:6379 (LISTEN)
redis-ser 649 Yuming    7u  IPv6 0x10d0674394866433      0t0  TCP localhost:6379 (LISTEN)

~% netstat | grep 6379
~% 

So you believe we don't need to run redis-server manually (at least with the software versions installed on my MacBook,  I noticed that starting redis-server manually IS necessary on another MacBook which was a few years older.)

Do you know who starts redis-server when MacBook boots?

The document https://github.com/rq/rq#getting-started still says that we must start redis-server by hand first.  Will the document be updated?

**Reply by ccrvlh** [reply]
Most likely `Homebrew` is starting the Redis Server.
I don't believe the doc needs to be updated - this is related to the user's setup, we are just making it clear that a running Redis Server is required (and it is).
